### ISSUE TITLE
Removing vpools concurrently failed with EOFError

### ISSUE THREAD

**Issue by JeffreyDevloo** [openvstorage/framework#1584]
### Problem description
While removing two vpools at the same time I received the same exception for both of them:
```
10.100.199.214 - ovs-workers.service: 2017-04-11 10:41:22 84500 +0200 - ovs04-211 - 3934/140477414532864 - celery/celery.worker.job - 211 - ERROR - Task ovs.storagerouter.remove_storagedriver[f066b9cb-08c6-4a22-b9ef-e7f7721fe0c8] raised unexpected: EOFError('connection closed by peer',)
10.100.199.214 - ovs-workers.service: Traceback (most recent call last):
10.100.199.214 - ovs-workers.service: File "/usr/lib/python2.7/dist-packages/celery/app/trace.py", line 240, in trace_task
10.100.199.214 - ovs-workers.service: R = retval = fun(*args, **kwargs)
10.100.199.214 - ovs-workers.service: File "/usr/lib/python2.7/dist-packages/celery/app/trace.py", line 438, in __protected_call__
10.100.199.214 - ovs-workers.service: return self.run(*args, **kwargs)
10.100.199.214 - ovs-workers.service: File "/opt/OpenvStorage/ovs/lib/storagerouter.py", line 1070, in remove_storagedriver
10.100.199.214 - ovs-workers.service: storagedriver_client.destroy_filesystem()
10.100.199.214 - ovs-workers.service: File "/usr/lib/python2.7/dist-packages/rpyc/core/netref.py", line 196, in __call__
10.100.199.214 - ovs-workers.service: return syncreq(_self, consts.HANDLE_CALL, args, kwargs)
10.100.199.214 - ovs-workers.service: File "/usr/lib/python2.7/dist-packages/rpyc/core/netref.py", line 71, in syncreq
10.100.199.214 - ovs-workers.service: return conn.sync_request(handler, oid, *args)
10.100.199.214 - ovs-workers.service: File "/usr/lib/python2.7/dist-packages/rpyc/core/protocol.py", line 438, in sync_request
10.100.199.214 - ovs-workers.service: self.serve(0.1)
10.100.199.214 - ovs-workers.service: File "/usr/lib/python2.7/dist-packages/rpyc/core/protocol.py", line 387, in serve
10.100.199.214 - ovs-workers.service: data = self._recv(timeout, wait_for_lock = True)
10.100.199.214 - ovs-workers.service: File "/usr/lib/python2.7/dist-packages/rpyc/core/protocol.py", line 345, in _recv
10.100.199.214 - ovs-workers.service: data = self._channel.recv()
10.100.199.214 - ovs-workers.service: File "/usr/lib/python2.7/dist-packages/rpyc/core/channel.py", line 50, in recv
10.100.199.214 - ovs-workers.service: header = self.stream.read(self.FRAME_HEADER.size)
10.100.199.214 - ovs-workers.service: File "/usr/lib/python2.7/dist-packages/rpyc/core/stream.py", line 195, in read
10.100.199.214 - ovs-workers.service: raise EOFError("connection closed by peer")
10.100.199.214 - ovs-workers.service: EOFError: connection closed by peer
```

The services have been removed.
```
root@ovs02-211:~# systemctl list-units | grep volumedriver
ovs-watcher-volumedriver.service                                            loaded active running   Watcher for Open vStorage master services
```
```
OVS running processes
=====================

ovs-albaproxy_myvpool02_0           active  28603
ovs-albaproxy_myvpool02_1           active  28686
ovs-arakoon-config                  active  7831
ovs-arakoon-mybackend-abm           active  20677
ovs-arakoon-mybackend-global-abm    active  25759
ovs-arakoon-mybackend-global-nsm_0  active  26112
ovs-arakoon-mybackend-nsm_0         active  20996
ovs-arakoon-mybackend02-abm         active  23224
ovs-arakoon-mybackend02-nsm_0       active  23567
ovs-arakoon-ovsdb                   active  8013
ovs-arakoon-voldrv                  active  27220
ovs-scheduled-tasks                 active  8267
ovs-support-agent                   active  8372
ovs-volumerouter-consumer           active  8265
ovs-watcher-config                  active  2883
ovs-watcher-framework               active  8263
ovs-watcher-volumedriver            active  28452
ovs-webapp-api                      active  8268
ovs-workers                         active  8825


OVS non-running processes
=========================

```
On a side note: shrinking another vpool down from 4 storagedrivers to 1 gave no issues. Removing it finally gave of the same error again.

Vpools were in running state when they got removed.
Gui logged something along the lines of 'Not all remove code was executed'

#### Mapping
```
In [2]: for vp in VPoolList.get_vpools():
   ...:     print vp.guid
   ...:     print vp.name
   ...:     
24c527fe-1dbe-4b26-832a-011387093539
integrationtests-vpool
27a73b83-6fab-4d8c-907d-2ff6f30eba1c
myvpool01
604cf8d0-1c7f-426c-b759-2754353daddf
myvpool02

```
(Log collected using log_reader across all nodes)
OVS workers log: 
[ovs-workers.zip](https://github.com/openvstorage/framework/files/912724/ovs-workers.zip)
Volumedriver log:
[voldr.zip](https://github.com/openvstorage/framework/files/912770/voldr.zip)

### Packages
```
ii  blktap-openvstorage-utils                           2.0.90-2ubuntu5                amd64                          utilities to work with VHD disk images files
ii  libblktapctl0-openvstorage                          2.0.90-2ubuntu5                amd64                          Xen API blktapctl shared library (shared library)
ii  libvhd0-openvstorage                                2.0.90-2ubuntu5                amd64                          VHD file format access library
ii  libvhdio-2.0.90-openvstorage                        2.0.90-2ubuntu5                amd64                          Xen API blktap shared library (shared library)
ii  openvstorage                                        2.8.2-dev.1491812208.82c2ec2-1 amd64                          openvStorage
ii  openvstorage-backend                                1.8.2-dev.1491895499.02690fd-1 amd64                          openvStorage Backend plugin
ii  openvstorage-backend-core                           1.8.2-dev.1491895499.02690fd-1 amd64                          openvStorage Backend plugin core
ii  openvstorage-backend-webapps                        1.8.2-dev.1491895499.02690fd-1 amd64                          openvStorage Backend plugin Web Applications
ii  openvstorage-core                                   2.8.2-dev.1491812208.82c2ec2-1 amd64                          openvStorage core
ii  openvstorage-hc                                     1.8.2-dev.1491895499.02690fd-1 amd64                          openvStorage Backend plugin HyperConverged
ii  openvstorage-health-check                           3.3.0-dev.1491384822.80948dc-1 amd64                          Open vStorage HealthCheck
ii  openvstorage-sdm                                    1.7.2-dev.1491895592.6669f9b-1 amd64                          Open vStorage Backend ASD Manager
ii  openvstorage-webapps                                2.8.2-dev.1491812208.82c2ec2-1 amd64                          openvStorage Web Applications

```

**Comment by kvanhijf** [comment]
@JeffreyDevloo : Inspecting the code shows that its normal that the volumedriver service had already been stopped and removed at that point in the code
@redlicha : Any idea when `LocalStorageRouterClient.destroy_filesystem` can throw an EOFError?

**Comment by JeffreyDevloo** [comment]
### Update
Including proxy logs as the volumedriver one will not give us anymore insight.

[proxies.zip](https://github.com/openvstorage/framework/files/913057/proxies.zip)

**Comment by redlicha** [comment]
@kvanhijf : seeing that this involves RPyC: could the `EOFError` come from that?

**Comment by kvanhijf** [comment]
Executing this in a local context:
```
In [8]: path1
Out[8]: 'arakoon://config/ovs/vpools/24c527fe-1dbe-4b26-832a-011387093539/hosts/integrationtests-vpool8Msef832NceGqLJ6/config?ini=%2Fopt%2FOpenvStorage%2Fconfig%2Farakoon_cacc.ini'

In [26]: sd = LocalStorageRouterClient(path1)

In [27]: sd.destroy_filesystem()
/usr/bin/ovs: line 285:   682 Aborted                 (core dumped) ipython
root@ovs01-211:~#
```

**Comment by kvanhijf** [comment]
Problem is in volumedriver. Related ticket: https://github.com/openvstorage/volumedriver/issues/309

**Comment by JeffreyDevloo** [comment]
Fixed in the latest OSE 6.16.
